evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -322,7 +347,131 @@ fn evm_error(retdata: &str) -> Capture<(ExitReason, Vec<u8>), Infallible> {
+fn evm_create_error(retdata: &str) -> Capture<(ExitReason, Option<Address>, Vec<u8>), Infallible> {
+    Capture::Exit((
+        ExitReason::Revert(ExitRevert::Reverted),
+        None,
+        ethers::abi::encode(&[Token::String(retdata.to_owned())]),
+    ))
+}
+
 impl<'a, 'b, B: Backend, P: PrecompileSet> CheatcodeStackExecutor<'a, 'b, B, P> {
+    fn expected_revert(
+        &mut self,
+        res: ExpectRevertReturn,
+        expected_revert: Option<Vec<u8>>,
+    ) -> ExpectRevertReturn {
+        let call = res.is_call();
+        if let Some(expected_revert) = expected_revert {
+            let data;
+            match res {
+                ExpectRevertReturn::Create(Capture::Exit((
+                    ExitReason::Revert(_e),
+                    None,
+                    revdata,
+                ))) => {
+                    data = Some(revdata);
+                }
+                ExpectRevertReturn::Call(Capture::Exit((ExitReason::Revert(_e), revdata))) => {
+                    data = Some(revdata);
+                }
+                _ => {
+                    if call {
+                        return ExpectRevertReturn::Call(evm_error(
+                            "Expected revert call did not revert",
+                        ))
+                    } else {
+                        return ExpectRevertReturn::Create(evm_create_error(
+                            "Expected revert create did not revert",
+                        ))
+                    }
+                }
+            }
+            let final_res = if let Some(data) = data {
+                if data.len() >= 4 && data[0..4] == [8, 195, 121, 160] {
+                    // its a revert string
+                    let decoded_data =
+                        ethers::abi::decode(&[ethers::abi::ParamType::Bytes], &data[4..])
+                            .expect("String error code, but not actual string");
+                    let decoded_data = decoded_data[0]
+                        .clone()
+                        .into_bytes()
+                        .expect("Can never fail because it is bytes");
+                    if decoded_data == *expected_revert {
+                        if call {
+                            return ExpectRevertReturn::Call(Capture::Exit((
+                                ExitReason::Succeed(ExitSucceed::Returned),
+                                DUMMY_OUTPUT.to_vec(),
+                            )))
+                        } else {
+                            return ExpectRevertReturn::Create(Capture::Exit((
+                                ExitReason::Succeed(ExitSucceed::Returned),
+                                Some(Address::from_slice(
+                                    &hex::decode("0000000000000000000000000000000000000001")
+                                        .expect("valid address hex")
+                                        [0..20],
+                                )),
```

Comment:
```suggestion
                                Some(Address::from_str("0000000000000000000000000000000000000001").unwrap())
```

Can use from_str or `.parse().`

Reply:
hmm i tried that and at least rust-analyzer told me that method didnt exist for `H160`. will try with this with actual compiler

Reply:
Should be fine https://github.com/gakonst/ethers-rs/blob/568d9c8697bf6ba4944543a21c9456c0c11f0caf/ethers-signers/src/wallet/private_key.rs#L224-L225